src/python/pants/help/help_printer.py

```diff
@@ -100,74 +130,57 @@ def format_goal(name: str, descr: str) -> str:
             formatted_descr = "\n".join(description_lines)
             return f"{name}{formatted_descr}\n"
 
-        lines = [
-            f"\n{title}\n",
-            f"Use `{self._bin_name} help $goal` to get help for a particular goal.",
-            "\n",
-            *(
-                format_goal(name, description)
-                for name, description in sorted(goal_descriptions.items())
-            ),
-        ]
-        print("\n".join(lines))
-
-    def _print_all_help(self) -> None:
-        print(self._get_help_json())
-
-    def _print_options_help(self) -> None:
-        """Print a help screen.
-
-        Assumes that self._help_request is an instance of OptionsHelp.
-
-        Note: Ony useful if called after options have been registered.
-        """
-        help_request = cast(OptionsHelp, self._help_request)
-        # The scopes explicitly mentioned by the user on the cmd line.
-        help_scopes = set(help_request.scopes)
-        if help_scopes:
-            for scope in sorted(help_scopes):
-                help_str = self._format_help(scope, help_request.advanced)
-                if help_str:
-                    print(help_str)
-            return
-        else:
-            self._print_global_help(help_request.advanced)
-
-    def _print_global_help(self, advanced: bool):
-        print(f"Pants {pants_version()} https://pypi.org/pypi/pantsbuild.pants/{pants_version()}")
-        print("\nUsage:")
-        print(
-            f"  {self._bin_name} [option ...] [goal ...] [target/file ...]  Attempt the specified goals."
+        for name, description in sorted(goal_descriptions.items()):
+            print(format_goal(name, description))
+        specific_help_cmd = f"{self._bin_name} help $goal"
+        print(f"Use `{self.maybe_green(specific_help_cmd)}` to get help for a specific goal.\n")
+
+    def _print_global_help(self):
+        def print_cmd(args: str, desc: str):
+            cmd = self.maybe_green(f"{self._bin_name} {args}".ljust(50))
+            print(f"  {cmd}  {desc}")
+
+        print(f"\nPants {pants_version()}")
+        print("\nUsage:\n")
+        print_cmd(
+            "[option ...] [goal ...] [file/target ...]",
```

Comment:
You can also add options after the specs, which is often much more convenient.

```suggestion
            "[option ...] [goal ...] [file/target ...] [option ...]",
```

Reply:
Yeah, but only if there are specs, I think. Otherwise the options will be bound to the goal scope, which gets complicated to explain.

Reply:
Yeah, that's correct.